pkg/controller/operators/operator_controller.go

```diff
@@ -115,40 +115,55 @@ func (r *OperatorReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	log := r.log.WithValues("request", req)
 	log.V(1).Info("reconciling operator")
 
-	// Fetch the Operator from the cache
+	// Get the Operator
 	ctx := context.TODO()
+	create := false
+	name := req.NamespacedName.Name
 	in := &operatorsv1.Operator{}
 	if err := r.Get(ctx, req.NamespacedName, in); err != nil {
 		if apierrors.IsNotFound(err) {
-			log.Info("Could not find Operator")
-			r.unobserve(req.NamespacedName)
-			// TODO(njhale): Recreate operator if we can find any components.
+			create = true
+			in.SetName(name)
 		} else {
-			log.Error(err, "Error finding Operator")
+			log.Error(err, "Error requesting Operator")
+			return reconcile.Result{Requeue: true}, nil
 		}
+	}
 
-		return reconcile.Result{}, nil
+	if !create {
+		if rv, ok := r.getLastResourceVersion(req.NamespacedName); ok && rv == in.ResourceVersion {
+			log.V(1).Info("Operator is already up-to-date")
+			return reconcile.Result{}, nil
+		}
 	}
-	r.observe(req.NamespacedName)
 
 	// Wrap with convenience decorator
 	operator, err := r.factory.NewOperator(in)
 	if err != nil {
 		log.Error(err, "Could not wrap Operator with convenience decorator")
-		return reconcile.Result{}, nil
+		return reconcile.Result{Requeue: true}, nil
```

Comment:
I don't think an error like this is transient; more likely it would be a programming error. If that's the case, wouldn't requeuing just congest the queue with something that would never reconcile correctly?

related: if we agree that an error here would mean the program is incorrect, then is a panic more appropriate?

Reply:
this function (currently) can only error out when `in` is nil, which will never be the case here.